File: mesop/components/divider/divider.py
import mesop.components.divider.divider_pb2 as divider_pb
from mesop.component_helpers import (
  insert_component,
  register_native_component,
)


@register_native_component
def divider(*, key: str | None = None, inset: bool = False):
  """Creates a Divider component.

  Args:
    key: The component [key](../components/index.md#component-key).
    inset: Whether the divider is an inset divider.
  """
  insert_component(
    key=key,
    type_name="divider",
    proto=divider_pb.DividerType(inset=inset),
  )

File: mesop/components/divider/divider.ts
import {MatDividerModule} from '@angular/material/divider';
import {Component, Input} from '@angular/core';
import {
  Key,
  Type,
} from 'mesop/mesop/protos/ui_jspb_proto_pb/mesop/protos/ui_pb';
import {DividerType} from 'mesop/mesop/components/divider/divider_jspb_proto_pb/mesop/components/divider/divider_pb';
import {Channel} from '../../web/src/services/channel';

@Component({
  templateUrl: 'divider.ng.html',
  standalone: true,
  imports: [MatDividerModule],
})
export class DividerComponent {
  @Input({required: true}) type!: Type;
  @Input() key!: Key;
  private _config!: DividerType;

  constructor(private readonly channel: Channel) {}

  ngOnChanges() {
    this._config = DividerType.deserializeBinary(
      this.type.getValue() as unknown as Uint8Array,
    );
  }

  config(): DividerType {
    return this._config;
  }
}
